### PR TITLE
Fix GetEnumerator

### PR DESCRIPTION
**Pull request by philjdf**
Would close https://github.com/G-Research/ParquetSharp/issues/242

### CODE REVIEW

File: csharp.test/TestLogicalTypeRoundtrip.cs [hunk 1]
```diff
@@ -288,6 +288,49 @@ public static void TestNestedStructArray([Values(Repetition.Required, Repetition
             fileReader.Close();
         }
 
+        [Test]
+        public static void TestLargeArraysEnumerator()
+        {
+            CheckEnumerator(Enumerable.Range(0, 4100).ToArray());
+            CheckEnumerator(Enumerable.Range(0, 4100).Select(i => new[] {$"row {i}"}).ToArray());
```
Comment (adeboyed) [inline csharp.test/TestLogicalTypeRoundtrip.cs:295]: This test works on the assumption that the default BufferLength=4096. 
Might be worth explicitly setting that.

Comment (philjdf) [reply csharp.test/TestLogicalTypeRoundtrip.cs]: Good idea, specified the buffer length to use and added a comment.

**Review by adamreeve — APPROVED** [review]

**Comment by jgiannuzzi** [comment]
Given the ABI change in this PR, and given that 5.0.0-beta3 has now been out and stable for a while, I would like to release 5.0.0 first, and then release 6.0.0-beta1 immediately after with the upgrade to Arrow 6.0.0 _and_ this PR.

@philjdf @adamreeve @adeboyed what are your thoughts on that?

**Comment by adeboyed** [comment]
Could this PR be included in the 5.0.0 release since upgrading to Arrow 6 may potentially lead to extended testing to ensure that the new features are supported and any API changes are made.

**Comment by jgiannuzzi** [comment]
> Upgrading to Arrow 6 may potentially lead to extended testing to ensure that the new features are supported and any API changes are made.

If you are talking about ParquetSharp itself, then there are no API changes stemming from the upgrade to Arrow 6, and no new features. All the work has already been done in #230 and is just waiting for 5.0.0 stable to be released.
Let me know if you meant something else!

> Could this PR be included in the 5.0.0 release?

If you mean 5.0.0 stable, then I would be quite reluctant in having an ABI change included without releasing another beta first. We _could_ however include it in another 5.0.0 beta. This would push the work on the 6.0.0 series further however, which I'm not really keen on, especially as #68 and #229 would be fixed by it.

**Comment by jgiannuzzi** [comment]
@adeboyed could you please let me know today if my comment above makes sense to you — if yes, I'll go ahead and release 5.0.0 and 6.0.0-beta1 (@philjdf and @adamreeve already agreed offline with this approach)

**Comment by adeboyed** [comment]
Hi Jonathan, this approach makes sense to me.
As an aside, we should change the comment here: https://github.com/G-Research/ParquetSharp/blob/8ce59329580e01a0e98eb157a56df4a43cfefcfc/README.md?plain=1#L145 to reflect that this logic has made it into a release version.

**Comment by jgiannuzzi** [comment]
Good catch @adeboyed! I made that change in PR #246 and released 5.0.0 stable.

**Comment by jgiannuzzi** [comment]
The upgrade to Arrow 6.0.1 has also been merged (PR #247).

**Comment by jgiannuzzi** [comment]
[ParquetSharp 6.0.1-beta1](https://github.com/G-Research/ParquetSharp/releases/tag/6.0.1-beta1) has been released with this fix.

**Pull request merged**